quic: flaky TestHolePunching

```
  === RUN   TestHolePunching
      conn_test.go:49: using a Ed25519 key: 12D3KooWJumRYKX7kwutYuYz8XPRCZ8YdMS7LxvgBueaibf4sboy
      conn_test.go:49: using a Ed25519 key: 12D3KooWGYWZQwNxChzDt1XJqB7G1oKhDM2aiCBxuWfKNY8Vrmas
      conn_test.go:532: 
          	Error Trace:	conn_test.go:532
          	            				asm_amd64.s:1571
          	Error:      	An error is expected but got nil.
          	Test:       	TestHolePunching
          	Messages:   	didn't expect to accept any connections
      conn_test.go:553: 
          	Error Trace:	conn_test.go:553
          	Error:      	Condition never satisfied
          	Test:       	TestHolePunching
  --- FAIL: TestHolePunching (0.11s)

--- a/p2p/transport/quic/conn_test.go
+++ b/p2p/transport/quic/conn_test.go
@@ -541,6 +541,10 @@ func TestHolePunching(t *testing.T) {
 		require.NoError(t, err)
 		connChan <- conn
 	}()
+	// Make sure the server role (the dial on t2) has progressed far enough.
+	// If it hasn't created the hole punch map entry, the connection will be accepted as a regular connection,
+	// which would make this test fail.
+	time.Sleep(25 * time.Millisecond)
 	conn1, err := t1.Dial(
 		network.WithSimultaneousConnect(context.Background(), true, ""),
 		ln2.Multiaddr(),

